clash-lib/src/Clash/Netlist.hs

```diff
@@ -338,23 +339,23 @@ mkFunApp dst fun args = do
             let topName = genComponentName [] mkId fun
             return (env </> (Text.unpack topName) <.> "manifest")
         Just man <- readMaybe <$> liftIO (readFile manFile)
-        instDecls <- mkTopUnWrapper fun annM man (dstId,dstHWty)
-                       (zip argExprs argHWTys)
+        instDecls <- mkTopUnWrapper fun annM man (dstPortType,dstId,dstHWty)
+                       (zip3 (map toPortType fArgTys) argExprs argHWTys)
         return (argDecls ++ instDecls)
 
       | otherwise -> error $ $(curLoc) ++ "under-applied TopEntity"
     _ -> do
       normalized <- Lens.use bindings
       case HashMap.lookup (nameOcc fun) normalized of
         Just _ -> do
-          (_,Component compName compInps [snd -> compOutp] _) <- preserveVarEnv $ genComponent (nameOcc fun)
+          (_,Component compName compInps [snd -> (_, i, hwType)] _) <- preserveVarEnv $ genComponent (nameOcc fun)
```

Comment:
Why not call the pattern variable `o` instead of `i`? That prevents the weird `i_` later on.

Reply:
Right. `i` does make more sense as it is an identifier, but I must say I dislike one-letter variable names in general so I'll probably change it to something else entirely.